core/src/main/java/software/amazon/awssdk/core/regions/Region.java

```diff
@@ -120,10 +121,11 @@ private Region(String value) {
      * For example, the following conditions will always evaluated to true:
      * {@code Region.of("us-east-1") == Region.of("us-east-1")}, {@code Region.US_EAST_1 == Region.of("us-east-1")}.</p>
      *
-     * @param value The name of the region to load.
+     * @param value The name of the region to load. Can't be null, empty or blank
      * @return The region associated with the provided name.
      */
     public static Region of(String value) {
+        Validate.notBlank(value, "Region value can't be null, empty or blank.");
```

Comment:
Would this be better as `Validate.paramNotBlank(value, "region");`

Reply:
Oh - it doesn't exist - maybe it should

Reply:
What would paramNotBlank do that notBlank doesn't?